File: src/main/java/com/sequenceiq/cloudbreak/repository/InstanceGroupRepository.java
package com.sequenceiq.cloudbreak.repository;

import org.springframework.data.repository.CrudRepository;
import org.springframework.data.repository.query.Param;

import com.sequenceiq.cloudbreak.domain.InstanceGroup;

public interface InstanceGroupRepository extends CrudRepository<InstanceGroup, Long> {

    InstanceGroup findOne(@Param("id") Long id);

    InstanceGroup findOneByGroupName(@Param("groupName") String groupName);

}

File: core/src/main/java/com/sequenceiq/cloudbreak/repository/InstanceGroupRepository.kt
package com.sequenceiq.cloudbreak.repository

import org.springframework.data.repository.CrudRepository
import org.springframework.data.repository.query.Param

import com.sequenceiq.cloudbreak.domain.InstanceGroup

@EntityType(entityClass = InstanceGroup::class)
interface InstanceGroupRepository : CrudRepository<InstanceGroup, Long> {

    override fun findOne(@Param("id") id: Long?): InstanceGroup

    fun findOneByGroupNameInStack(@Param("stackId") stackId: Long?, @Param("groupName") groupName: String): InstanceGroup

}